toolchain/sem_ir/node.h

```diff
@@ -207,272 +208,329 @@ struct MemberIndex : public IndexBase, public Printable<MemberIndex> {
   }
 };
 
-// The standard structure for Node. This is trying to provide a minimal
-// amount of information for a node:
-//
-// - parse_node for error placement.
-// - kind for run-time logic when the input Kind is unknown.
-// - type_id for quick type checking.
-// - Up to two Kind-specific members.
-//
-// For each Kind in NodeKind, a typical flow looks like:
-//
-// - Create a `Node` using `Node::Kind::Make()`
-// - Access cross-Kind members using `node.type_id()` and similar.
-// - Access Kind-specific members using `node.GetAsKind()`, which depending on
-//   the number of members will return one of NoArgs, a single value, or a
-//   `std::pair` of values.
-//   - Using the wrong `node.GetAsKind()` is a programming error, and should
-//     CHECK-fail in debug modes (opt may too, but it's not an API guarantee).
-//
-// Internally, each Kind uses the `Factory*` types to provide a boilerplate
-// `Make` and `Get` methods.
-class Node : public Printable<Node> {
- public:
-  struct NoArgs {};
+// Data storage for the operands of each kind of node.
```

Comment:
Does it make sense to declare (and maybe use) these with the `.def` file to ensure that every one is covered, even if with a single line to make it an explicit alias?

Or will the failure of a missing node be obvious and caught already?

Could maybe expand this comment to hint a bit about how these are used or checked for consistency?

Reply:
After this section we do use the `.def` file to inject typed node wrappers for each node declared in the `.def` file, so missing nodes here will get caught, with an error saying that type `NodeData::YourNewNode` doesn't exist, which seems like it should be reasonably obvious. Missing nodes in the `.def` file will get caught because the typed node type won't exist (which might be a somewhat less obvious error).